setupVars.c

```diff
@@ -158,6 +158,11 @@ bool insetupVarsArray(char * str)
 				free(domain);
 				return true;
 			}
+			else
+			{
+				free(domain);
+				return false;
```

Comment:
This is wrong, it is not possible to return here since that would be skipping the `for`-loop altogether. Since I don't have write permissions on your fork, I'll open a new PR for this.